File: Codes_morgan/TOURE/Pb_inverse/2.Illustrations-Python-et-Matlab/2.Illustrations-Python-et-Matlab/Old/3.Fourier-Debruit-Filtres/Fourier.m

clear all ; close all ;

pause; % onde pure amortie
t = (0:1/512:1);
f = exp(-t.^2*4).* cos(20*(2*pi*t));
figure; ;subplot(1,2,1); plot(t(1:512),f(1:512)); title('SINUS AMORTI') ;
pause;
F = abs(fft(f,512));
subplot(1,2,2); plot(t(1:128),0,t(1:128),F(1:128)); title('TRANSFORMEE DE FOURIER') ;


pause; % onde pure amortie plus rapide
t = (0:1/512:1);

f = exp(-t.^2*4).* cos(60*(2*pi*t));
figure; subplot(1,2,1); plot(t(1:512),f(1:512)); title('...EN PLUS RAPIDE') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:128),0,t(1:128),F(1:128)); title('TRANSFORMEE DE FOURIER') ;

pause; % somme de 2 ondes pures
t = (0:1/512:1);
f = 2*cos(2*(2*pi*t))+cos(20*(2*pi*t));
figure; subplot(1,2,1); plot(t(1:512),f(1:512)); title('SOMME DE DEUX SINUS') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:32),0,t(1:32),F(1:32)); title('TRANSFORMEE DE FOURIER') ;

pause; % ondes pure + bruit
t = (0:1/512:1);
f = cos(3*(2*pi*t))+ randn(size(t));
figure; subplot(1,2,1); plot(t(1:512),f(1:512));title('SINUS AVEC BRUIT') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:32),0,t(1:32),F(1:32)); title('TRANSFORMEE DE FOURIER') ;

pause; % produit de 2 ondes pures
t = (0:1/512:1);
f = cos(1*(2*pi*t)).*cos(30*(2*pi*t));
figure; subplot(1,2,1); plot(t(1:512),f(1:512)); title('PRODUIT DE DEUX SINUS') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:64),0,t(1:64),F(1:64)); title('TRANSFORMEE DE FOURIER') ;


pause; % porte 
t = (0:1/512:1);
% f = (1-abs(t)).*square(12*t);
f = rectpuls(5*t);
figure; subplot(1,2,1); plot(t(1:512),f(1:512)); title('PORTE') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(F(1:128)); title('TRANSFORMEE DE FOURIER') ;

pause; % chirp "chaotique"... bruit artificiel
t = (0:1/32:10);
f = cos(10*t.^2);
figure; subplot(1,2,1); plot(t(1:320),f(1:320)); title('GAZOUILLIS') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:80),0,t(1:80),F(1:80)); title('TRANSFORMEE DE FOURIER') ;

pause;  % gaussienne
t = (0:1/512:1);
f = exp(-(t-.5).^2*20);  
figure; subplot(1,2,1); plot(t(1:512),f(1:512)); title('GAUSSIENNE') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:32),0,t(1:32),F(1:32)); title('TRANSFORMEE DE FOURIER') ;

pause; % gaussienne plus contractée
t = (0:1/512:1);
f = exp(-(t-.5).^2*100);
figure; subplot(1,2,1); plot(t(1:512),f(1:512)); title('GAUSSIENNE') ;
pause;
F = abs(fft(f));
subplot(1,2,2); plot(t(1:32),0,t(1:32),F(1:32)); title('TRANSFORMEE DE FOURIER') ;

File: Codes_morgan/TOURE/Pb_inverse/2.Illustrations-Python-et-Matlab/2.Illustrations-Python-et-Matlab/3._TF-convolution/Fourier.py
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
#from math import *
from numpy import *
#from scipy import * 
from matplotlib import * 
from matplotlib.pyplot import *
from numpy.fft import fft, ifft
#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

print("",end='\n\n') 
print("Quelques transformées de Fourier", end='\n\n')

#################################################################
print("----------- Onde pure amortie --------------", end='\n\n') 
t = linspace(0,1,512) 
f = exp(-t**4)*cos(10*(2*pi*t))
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Onde pure amortie plus rapide --------------", end='\n\n') 
t = linspace(0,1,512) 
f = exp(-t**4)*cos(20*(2*pi*t))
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Somme de deux ondes pures --------------", end='\n\n') 
t = linspace(0,1,512) 
f = 2*cos(2*(2*pi*t)) + cos(20*(2*pi*t)) 
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Onde pure avec bruit additif --------------", end='\n\n') 
t = linspace(0,1,512) 
f = cos(3*(2*pi*t)) + random.normal(0, 1/2, 512)
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Produit de deux sinusoïdes--------------", end='\n\n') 
t = linspace(0,1,512) 
f =  cos(1*(2*pi*t)) * cos(30*(2*pi*t))
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Fonction porte --------------", end='\n\n') 
t = linspace(0,1,512) 
f = piecewise(t, [t >= 1/3, t >= 2/3 ], [1, 0])
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Bruit artificiel (Chirp) --------------", end='\n\n') 
t = linspace(0,1,512) 
f = cos(500*t**2+200*t**3)
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")


#################################################################
print("----------- Gaussienne --------------", end='\n\n') 
t = linspace(0,1,512) 
f = exp(-5000*(t-.5)**2)
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")

#################################################################
print("----------- Gaussienne plus contractée --------------", end='\n\n') 
t = linspace(0,1,512) 
f = exp(-10000*(t-.5)**2)
F = abs(fft(f))
fig, (ax1, ax2) = subplots(2, 1)
ax1.plot(t,f) 
ax1.set_title('Signal f')
ax2.plot(t[1:256],F[1:256],'r') 
ylabel('Amplitude')
ax2.set_title('Transformée de Fourier F')
show()
#input("----")